docs-infra: Remove `rev:true` property

sonukapoor opened the pull request:
This property was used when the data was stored in the `Firebase` database
and `rev:true` meant that an admin has reviewed and approved the data. It
is however not used anymore.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

gkalpak reviewed (changes requested):
Thx, @sonukapoor! BTW, you also need to remove `rev` from the other places it is used. Eg.:
- [resource-list.component.html](https://github.com/angular/angular/blob/2d5bb26b028397990f0dcd677bf5c27f57be088a/aio/src/app/custom-elements/resource/resource-list.component.html#L16)
- [resource.model.ts](https://github.com/angular/angular/blob/39e066add18c48af24db80ee00b775d73d926dab/aio/src/app/custom-elements/resource/resource.model.ts#L20)
- [resource.service.spec.ts](https://github.com/angular/angular/blob/d349cd91b162336515f413c81e0dc9afcf1cc067/aio/src/app/custom-elements/resource/resource.service.spec.ts)

##
For context:
The `rev` property has been in the initial commit that introduced `resource.json` (196203f6d76ac5240d1485dbd9701bfb026eab56) and has been added in all new entries since (always with value `true`). IIRC, this field is a remnant from back when this data was stored in a Firebase Database and the `rev` field indicated whether the entry has been reviewed/approved by a DevRel lead or something. Now that the data is kept in the repository and the reviewing is done as part of the corresponding PR, this field is no longer necessary.

@sonukapoor, can you please add that info in the commit message ☝️

gkalpak commented:
BTW, I believe the commit should be of type `docs-infra`, not `docs`.

sonukapoor commented on aio/content/marketing/resources.json:
```diff
@@ -713,132 +612,112 @@
     }
   }
-}
+}
```
Fixed in https://github.com/angular/angular/pull/37181/commits/2491746815f3e31ae0425acb28628bfb8aa3f472

sonukapoor commented on aio/content/marketing/resources.json:
```diff
@@ -7,38 +7,32 @@
             "url": "https://www.reddit.com/r/Angular2/"
           },
-          "angular-devto" : {
+          "angular-devto": {
```
Thanks @gkalpak - I have made the changes to the component / model and spec file. 

https://github.com/angular/angular/pull/37181/commits/2491746815f3e31ae0425acb28628bfb8aa3f472

IgorMinar reviewed (approved):
LGTM! Thank you!

Reviewed-for: global-approvers

sonukapoor committed:
refactor(docs-infra): remove `rev` property from `Resource` interface and data

The `rev` property has been in the initial commit that introduced
`resources.json` (196203f6d76ac5240d1485dbd9701bfb026eab56) and has been
added to all new entries since (always with the value `true`). This
field is a remnant from back when this data was stored in a Firebase
Database and the `rev` field indicated whether the entry has been
reviewed/approved by a DevRel lead or something. Now that the data is
kept in the repository and the reviewing is done as part of the
corresponding PR, this field is no longer necessary.

sonukapoor committed:
fixup! refactor(docs-infra): remove `rev` property from `Resource` interface and data

gkalpak reviewed (approved):
Thx, @sonukapoor ✨ 

FYI, I've massaged the commit message a bit and rebased on master.

sonukapoor commented:
Thank you @gkalpak
